features/filesystem/fat/FATFileSystem.cpp

```diff
@@ -160,6 +160,24 @@ void ff_memfree(void *p)
 }
 
 // Implementation of diskio functions (see ChaN/diskio.h)
+static WORD disk_get_sector_size(BYTE pdrv)
+{
+    WORD ssize = _ffs[pdrv]->get_erase_size();
+    if (ssize < 512) {
+        ssize = 512;
+    }
+
+    MBED_ASSERT(ssize >= _MIN_SS && ssize <= _MAX_SS);
```

Comment:
The block device needs to be initialized before `get_erase_size` and friends can be called reliably. Currently the init is called from the chan layer during mount, but mount may also write to the device.

We could bring up the block device to check things before mounting, but since these are asserts anyways it's easier to just put them in the chan-disk layer.

The assert for `is_valid_erase` should live in the block device driver.